package NClusterMaster;

import "library/cpp/getoptpb/proto/confoption.proto";

message TSolverOptions {
    optional bool DoNotDaemonize = 1        [(NGetoptPb.Conf) = {Short: 'f', Long: "do-not-daemonize", Descr: "do not daemonize"},
        default = false];
    optional uint32 Port = 2                [(NGetoptPb.Conf) = {Short: 'p', Long: "port", Descr: "listen port for incoming connections",
        Type: "port"}, default = 3366];
    optional string Path = 3                [(NGetoptPb.Conf) = {Short: 'u', Long: "path", Descr: "listen path for incoming connections",
        Type: "PATH"}];
    optional string PidFile = 4             [(NGetoptPb.Conf) = {Short: 'P', Long: "pidfile", Descr: "pidfile", Type: "PATH"}];
    optional string Logfile = 5             [(NGetoptPb.Conf) = {Short: 'l', Long: "logfile", Descr: "logfile", Type: "PATH"}];
    optional uint32 HTTPPort = 6            [(NGetoptPb.Conf) = {Short: 'h', Long: "http-port", Descr: "solver http port", Type: "port"},
        default = 3132];
    optional bool RetryListen = 7           [(NGetoptPb.Conf) = {Short: 'r', Long: "retry", Descr: "retry create listeners until success"},
        default = false];
    optional string KnownLimitsFile = 8     [(NGetoptPb.Conf) = {Short: 'k', Long: "known-limits-file", Descr: "read known limits from file",
        Type: "path"}];
    optional uint32 FailureDelaySeconds = 9 [(NGetoptPb.Conf) = {Short: 'd', Long: "failure", Descr: "failure delay in seconds"},
        default = 1];
    optional uint32 ListenersBacklog = 10   [(NGetoptPb.Conf) = {Short: 'b', Long: "listeners-backlog", Descr: "listeners backlog"},
        default = 4096];
}
